src/server/src/main/java/io/cassandrareaper/resources/RepairRunResource.java

```diff
@@ -379,22 +393,33 @@ public Response modifyRunState(
           .entity("invalid \"state\" argument: " + state.get())
           .build();
     }
-    final RepairRun.RunState oldState = repairRun.get().getRunState();
+    final RepairRun.RunState oldState = repairRun.getRunState();
+
+    if (intensityStr.isPresent() && oldState == newState && newState == RepairRun.RunState.PAUSED) {
+      final Response intensityValidationResponse = validateIntensity(intensityStr);
+      if (intensityValidationResponse != null) {
+        return intensityValidationResponse;
+      }
+
+      final double intensity = Double.parseDouble(intensityStr.get());
+      return editRun(repairRun, repairUnit, segmentsRepaired, intensity);
```

Comment:
> > what's the response to the (restful) user when newState != RepairRun.RunState.PAUSED a change in intensity was specified but not fulfilled but the repair run was otherwise modified?
>
> intensity parameter will be ignored in case when old state and new state is not PAUSED.
> So for next two requests
> `PUT state=RUNNING&intensity=null`
> `PUT state=RUNNING&intensity=0.7`
> response will be `Not Modified`
> Just like intensity parameter doesn't exist.
> For this case can be more aggressive logic like `Bad Request`. But I don't understand profit of this logic.

I don't think we need more return codes or logic, eg `Bad Request`.
It's just to review that we arn't going to be returning any response status codes that confuse users…

So the logic is (in order of precedence): 
 
 1. if request does not specify new state, then BAD_REQUEST is returned,
 1. if the repair run (or unit) is not found, then NOT_FOUND is returned,
 1. if there's any other repair runs for the related repair unit and they are RUNNING, then return CONFLICT,
 1. if state is unaltered and PAUSED, intensity specified but invalid then BAD_REQUEST is returned,
 1. if state is unaltered and PAUSED, intensity specified and valid then OK is returned,
 1. if state is unaltered, all other parameters are ignored, and NOT_MODIFIED is returned,
 1. if state is altered, but an invalid transition, then all other parameters are ignored, and METHOD_NOT_ALLOWED returned,
 1. if state is altered, the transition valid, then all other parameters are ignored, and OK returned.

Does this sound right?
The issue I see here is that there's no catch of when:
 
 - (in step 5) intensity is specified and valid but matches existing value. NOT_MODIFIED should be returned instead of OK.
 - (after step 5) when a state transition is specified as well as a new intensity but that new intensity is ignored.

What do you think about creating a separate REST endpoint for handling updates to intensity, so to simplify the above logic?
I'm open to ideas…

Reply:
I thought about another endpoint but current logic means
PUT repair_run/{id} -- edit repair run with {id}
There is nothing about edit only state.
That's why I wrote code in this method.
In future you perhaps will want to extend edit (PUT) method by another parameters. Think about it. How would you like to do it?

If you want my opinion I think that this method is messy. Better to rewrite all method transitions to state machine implementation. So there will be just one PUT method but it will reroute call through finite state machine graph.